core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java

```diff
@@ -32,6 +32,7 @@
  * Prefer implementing this interface over implementing {@link Streamable} where possible. Lots of code depends on {@linkplain Streamable}
  * so this isn't always possible.
  */
+@FunctionalInterface
```

Comment:
I don't think Writeable is intended to be used as a reference to a function. It is intended to be implemented. You can use it as a functional reference in rare cases but I don't think this interface is a good thing here.